llvm/lib/Target/AArch64/AArch64MIPeepholeOpt.cpp

```diff
@@ -715,6 +719,58 @@ bool AArch64MIPeepholeOpt::visitFMOVDr(MachineInstr &MI) {
   return true;
 }
 
+bool AArch64MIPeepholeOpt::visitUBFMXri(MachineInstr &MI) {
+  // Check if the instruction is equivalent to a 32 bit LSR or LSL alias of
+  // UBFM, and replace the UBFMXri instruction with its 32 bit variant, UBFMWri.
+  int64_t Immr = MI.getOperand(2).getImm();
+  int64_t Imms = MI.getOperand(3).getImm();
+
+  bool IsLSR = Imms == 31 && Immr <= Imms;
+  bool IsLSL = Immr == Imms + 33;
+  if (IsLSR || IsLSL) {
```

Comment:
LLVM usually prefers to return early, lowering the indentation level.
```
if (!LSR && !IsLSL)
  return false;
```

Reply:
Thanks for catching that, fixed it now.